SharpGenTools.Sdk/Documentation/DocProviderExecutor.cs

```diff
@@ -0,0 +1,299 @@
+                                    }
+
+                                    nextDelay = backoff[backoffIndex++];
+
+                                    if (nextDelay < retryDelay)
+                                        (backoff, backoffIndex, nextDelay) = GenerateBackoff(retryDelay);
+
+                                    break;
+                                }
+                                case FindDocumentationResultSuccess resultSuccess:
+                                    return resultSuccess.Item; // TODO: check if the item is empty (therefore, useless)
+                                default:
+                                    throw new ArgumentOutOfRangeException(
+                                        nameof(result),
+                                        $"Unexpected {nameof(IFindDocumentationResult)}: {result.GetType().FullName}"
+                                    );
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            e.Data["SDK:" + nameof(docProvider)] = docProvider;
+                            e.Data["SDK:" + nameof(docName)] = docName;
+                            e.Data["SDK:" + nameof(context)] = context;
+                            e.Data["SDK:" + nameof(retry)] = retry;
+                            e.Data["SDK:" + nameof(backoffIndex)] = backoffIndex;
+                            e.Data["SDK:" + nameof(exceptions) + ".Count"] = exceptions.Count;
+
+                            exceptions.Add(e);
+
+                            // We should retry less when it's due to unhandled exception.
+                            // So in exception case we step twice in retry count on each iteration.
+                            retry++;
+                        }
+                    }
+
+                    context.Logger.Message($"{docProviderName.Value} extension failed to find documentation for \"{docName}\"");
+
+                    return null;
+                }
+                finally
+                {
+                    if (exceptions.Count > 0)
+                    {
+                        var failure = new DocumentationQueryFailure(docName)
+                        {
+                            Exceptions = exceptions,
+                            FailedProviderName = docProviderName.Value,
+                            TreatProviderFailuresAsErrors = docProvider.TreatFailuresAsErrors
+                        };
+
+                        context.Failures.Add(failure);
+                    }
+                }
+            }
+        }
+
+        private static (TimeSpan[] backoff, int index, TimeSpan? nextDelay) GenerateBackoff(TimeSpan offset) =>
+            // BUG: 5 should be sufficient, but causes IndexOutOfBounds
+            (Backoff.DecorrelatedJitterBackoffV2(offset + TenthOfSecond, 6).ToArray(), 0, null);
```

Comment:
Note that this backoff formula can produce `TimeSpan`s like (_mm:ss_, for `offset` = _00:10_): `00:11, 00:26, 03:01, 00:48, ...`.